control-plane/controllers/configentry_controller.go

```diff
@@ -422,13 +439,14 @@ func assignServiceVirtualIP(ctx context.Context, logger logr.Logger, consulClien
 		return err
 	}
 
+	consulType := configEntry.ToConsul(datacenter)
```

Comment:
I like this change. It was a spot that just didn't feel right to me when I did it...